### ISSUE TITLE
Allow non-literal (but const) keys

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#16]
This shouldn't be flagged:
```
const arr = [0, 1, 2] as const; const key = 0 as const; const foo = arr[key];
```

**Comment by danielnixon** [comment]
@tankers746 PRs welcome ;-)

**Comment by danielnixon** [comment]
Fixed in https://github.com/danielnixon/eslint-plugin-total-functions/commit/00a939935569510d6b92603b2d894f6c99aea531
